packages/core/src/helpers/web.ts

```diff
@@ -9,3 +9,22 @@ export function injectScript(win: Window, rootComId: string, scriptUrl: string)
         win.document.head.appendChild(scriptEl);
     });
 }
+
+export function getEngineEntryOptions(envName: string) {
```

Comment:
I know it wasn't there before but can you add a simple unit test to this function it has like 3 layer fallback that I would like to make sure is in tact.

my suggestion is to pass the `globalThis` with a narrow type of all the required fields for this function to work.

Also it's not safe to split `url` string by `?` you could use `URL.parse` and get the `.searchParams` from there.

at the end this function can return a regular `Map` and not `URLSearchParams` I did that because I only had that in hand already. it will make the final loop cleaner by checking the same map keys no additional set is needed